openpgp.d.ts

```diff
@@ -22,6 +22,8 @@ export function generateSessionKey(options: { encryptionKeys: PublicKey[], date?
 export function decryptKey(options: { privateKey: PrivateKey; passphrase?: string | string[]; config?: PartialConfig }): Promise<PrivateKey>;
 export function encryptKey(options: { privateKey: PrivateKey; passphrase?: string | string[]; config?: PartialConfig }): Promise<PrivateKey>;
 export function reformatKey(options: { privateKey: PrivateKey; userIDs?: UserID|UserID[]; passphrase?: string; keyExpirationTime?: number; config?: PartialConfig }): Promise<KeyPair>;
+export function revokeKey(options: { key: PrivateKey, reasonForRevocation?: { flag?: enums.reasonForRevocation, string?: String }, config?: PartialConfig }): Promise<PrivateKey>;
+export function revokeKey(options: { key: PublicKey, revocationCertificate?: String, config?: PartialConfig }): Promise<PublicKey>;
```

Comment:
Thanks, and sorry for the delay! However, I think this return type currently doesn't match what the function actually returns - it either returns a `KeyPair`, or a partial one. That being said, I think it's kind of ugly and this type definition would make more sense, and would make the function simpler - so if you want, you could make a separate PR to just return the new key object, rather than a wrapper object with the armored keys etc?

Reply:
I agree and opened #1349 based on this comment. Let me know if you were thinking something different.